### ISSUE TITLE
Support Hive's TIMESTAMP WITH LOCAL TIME ZONE

### ISSUE THREAD

**Issue by findepi** [prestosql/presto#1240]
Hive 3.1 adds a new data type: `TIMESTAMP WITH LOCAL TIME ZONE`. 

Example:
```
beeline> create table x(t timestamp, tltz timestamp with local time zone);
OK

beeline> desc x;
+-----------+---------------------------------------------------+----------+
| col_name  |                     data_type                     | comment  |
+-----------+---------------------------------------------------+----------+
| t         | timestamp                                         |          |
| tltz      | timestamp with local time zone('Asia/Kathmandu')  |          |
+-----------+---------------------------------------------------+----------+
```

Add support for this type in Hive connector.

**Comment by dain** [comment]
We should map this to `TIMESTAMP WITH TIME ZONE`

**Comment by martint** [comment]
I agree that we should map this to `TIMESTAMP WITH TIME ZONE` with a fixed timezone that is insensitive to daylight savings or other adjustments that lead to ambiguities (i.e., practically speaking, UTC). This is the best we can do to provide interoperability with Hive for this type. The alternative is not to support it at all, which means many Hive users cannot easily switch to Trino or query their data.

**Comment by findepi** [comment]
if we had "instant" type (https://github.com/trinodb/trino/issues/2273) we could map to that.

**Comment by mosabua** [comment]
In order to solve this problem I suggest we review this PR and merge it with the current proposed approach to unblock existing users. If we ever end up creating an instant type we can reconsider and change.
